tools/mo/openvino/tools/mo/back/preprocessing.py

```diff
@@ -70,7 +70,9 @@ def check_keys_valid(ov_function: Model, keys: list, search_outputs: bool):
     if search_outputs:
         nodes += ov_function.outputs
 
-    for name in keys:
+    rename_dict = {}
+
+    for name in dict_to_validate.keys():
```

Comment:
I propose to do this within two steps:
1) rename keys in dictionary if needed
2) perform this check

Reply:
Also, please add comments for each pass about why it is needed